### DISCUSSION TITLE
Changing Forests upon reload

### DISCUSSION THREAD

**Discussion by Mephisto-Fibs** [Keriew/augustus#539] [Q&A] [answered]
I discovered that since version 3.1 the forest changes with existing maps when a road is drawn through the trees. The problem arises when you save a map and reload it. This is especially a problem with custom maps, as the map changes visually and is therefore at least less aesthetically pleasing.

Was there a change that caused this?

**Comment by crudelios** [comment]
Yes there was, and I was unaware of this. I'll have to check out a way to fix this. I'm moving this to the issues tab.

**Comment by Mephisto-Fibs** [comment]
Hey there :)

just wanted to let you know that this issue is still there, even in the latest unstable build (3.1.0.54).

**Reply by crudelios** [reply]
~Are you sure? Old savefiles will not recover the forests, only new ones will.~

Scratch that. I just figured out a way to recover forest info from *some* savefiles. Not all of them will work though, and for those there's nothing I can do.

**Reply by Mephisto-Fibs** [reply]
Thank you :)

*tested it with a fresh world with forests intact... drew a street through it, saved and upon reload the forest's density was gone.

**Reply by crudelios** [reply]
Can you post the save before and after drawing the road please?

**Comment by Mephisto-Fibs** [comment]
[Map bevore and after save.zip](https://github.com/Keriew/augustus/files/7750012/Map.bevore.and.after.save.zip)

I created this map (today, using the newest unstable build) just to test the forest issue ... whenever a map is created into a .svx the forest's density becomes lighter. 

Furthermore, also, upon loading a .svx (that was unaffected before) the forest is loaded in it's less dense appearance. 
Here is the map (that is loaded fine but after saving and reloading has lost it's forests)
I also included a save from which I know that it is unafected by this new bug.... upon loading this map it also appears with less dense forests. I hope this helps.
[Other saves.zip](https://github.com/Keriew/augustus/files/7750103/Other.saves.zip)

**Comment by crudelios** [comment]
How about this:

Try loading a save (not a map) which has dense forests, draw a road in the middle of a forest then reload. That's the only use case I fixed.

**Comment by Mephisto-Fibs** [comment]
A road isn’t even needed as the act of creating a save alone leads to a diminishing density of the forest areas. Even loading up a previous save (pre 3.1) in 3.1 does it; the save loads with a very lightly packed forest where there was a lush thick forest before (as shown in the second upload of Augusta Vindelicum; there is the original map file, an unaffected save with lush forests from before 3.1 and a save I did yesterday…. )

Its as if the forest draws up itself again; as if in the map creating process, where a thick forest is only possible if it covers a relatively huge and uninterrupted area.

**Idea:
Maybe its the image id recalculation upon city load that was newly added in 3.1 that causes the forests to change? Because after loading, there are actually not less trees! In fact, there are as many trees as before but in their smaller, more shrub-like appearance making the forest seem lighter.
If this is the case, please reverse this change or exclude trees from the image id recalculation; … or is this a desired effect? I have to say that I liked my roads traversing through dense foliage…**


Or, maybe my game corrupted somehow after updating to 3.1?


Can someone else test this in their game, too?

**Comment by crudelios** [comment]
It's definitely the image id recalculation that causes this, and this won't be reverted as the change is needed moving forward.

I'll try to mitigate the issue though. Some existing saves can't recover the forests but some will, and new saves will also properly count the trees when I'm done.

To be clear: saves from before v3.1 will recover the trees, as will all scenario files. Saves after I fix this will also keep the trees.

The only unrecoverable saves are the ones created between v3.1 and the change I'll implement.

**Comment by Mephisto-Fibs** [comment]
Hey there, here is the annoying forest guy again :P 

I am sure you know this but I just wanted to comment that it is not the number of trees that are changing, it is their appearance. 
Is there any way to make the model of each tree on the map look like the model that was there at the start of the map? Or in other words, saving and reloading shouldn't change the trees model/ image id ... is that possible?

**Comment by crudelios** [comment, marked as answer]
This should finally be fixed!

However, the fix will not work for savegames created after v3.1.0 and before today. In those files, the original tree density information is already lost and cannot be recovered anymore.

Still, the bug won't happen anymore, which is good!

**Comment by Mephisto-Fibs** [comment]
Great news! I cannot wait to test this in a new update. Thank you so much. This will make mapbuilding less frustrating and more reliable.
